app/database/operator/server_data_operator/transformers/category.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {MM_TABLES} from '@constants/database';
+import {prepareBaseRecord} from '@database/operator/server_data_operator/transformers/index';
+import {OperationType} from '@typings/database/enums';
+
+import type {TransformerArgs} from '@typings/database/database';
+import type CategoryModel from '@typings/database/models/servers/category';
+import type CategoryChannelModel from '@typings/database/models/servers/category_channel';
+
+const {
+    CATEGORY,
+    CATEGORY_CHANNEL,
+} = MM_TABLES.SERVER;
+
+/**
+ * transformCategoryRecord: Prepares a record of the SERVER database 'Category' table for update or create actions.
+ * @param {TransformerArgs} operator
+ * @param {Database} operator.database
+ * @param {RecordPair} operator.value
+ * @returns {Promise<CategoryModel>}
+ */
+export const transformCategoryRecord = ({action, database, value}: TransformerArgs): Promise<CategoryModel> => {
+    const raw = value.raw as Category;
+    const record = value.record as CategoryModel;
+    const isCreateAction = action === OperationType.CREATE;
+
+    // id of category comes from server response
+    const fieldsMapper = (category: CategoryModel) => {
+        category._raw.id = isCreateAction ? (raw?.id ?? category.id) : record.id;
+        category.displayName = raw.display_name;
+        category.sorting = raw.sorting;
+        category.sortOrder = raw.sort_order / 10;
```

Comment:
Please describe magic number 10 here or move into a constant. Perhaps in `@constant/database`
``` 
        category.sortOrder = raw.sort_order / 10;  
```